client-v2/src/util/CAPIUtils.ts

```diff
@@ -4,7 +4,9 @@ import { ArticleFragment } from '../shared/types/Collection';
 
 const getIdFromURL = (url: string): string | null => {
   const [, id = null] =
-    url.match(/^https:\/\/www.theguardian\.com\/(.*)\??/) || [];
+    url.match(
+      /^https:\/\/(?:www.theguardian\.com|viewer.gutools.co.uk)\/(?:(?:preview|live)\/)?(.*)\??/
```

Comment:
Currently this will return the wrong path for any articles that exist at a path such as https://www.theguardian.com/live/myarticle - it will only return `myarticle` for the path as live will match the regex and not be captured.

I just checked and currently https://www.theguardian.com/live redirects to https://www.theguardian.com/tone/minutebyminute so perhaps not an issue but could be future-proofed with:

```suggestion
      /^https:\/\/(?:www.theguardian.com|viewer.gutools.co.uk\/(?:preview|live))\/([^?]*)/
```

If this seems worthwhile then maybe we can add a test for it?

Reply:
I'm not 100% of all the URLs but have tweaked the regex slightly to match all of the below:

```
/^https:\/\/(?:www.theguardian.com\/|viewer.gutools.co.uk(?:\/(?:preview|live))?\/)([^?]*)/
```

![image](https://user-images.githubusercontent.com/32312712/47805781-124cec80-dd30-11e8-9fea-38908175c42f.png)

Reply:
Awesome - sorry if mine missed any of your test cases!

Reply:
Also I just wanted to try the new github suggested edits 🙄

Reply:
New github edits  👍  If this now covers everything I'll merge the PR after rebasing